app-emulation/fs-uae/patches/fs_uae-2.8.3.patchset

```diff
@@ -0,0 +1,503 @@
+index bdbda8c..21db919 100644
+--- a/manymouse/x11_xinput2.c
++++ b/manymouse/x11_xinput2.c
+@@ -10,7 +10,7 @@
+ #include "manymouse.h"
+ /* Try to use this on everything but Windows and Mac OS by default... */
+ #ifndef SUPPORT_XINPUT2
+-#if ( (defined(_WIN32) || defined(__CYGWIN__)) )
++#if ( (defined(_WIN32) || defined(__CYGWIN__) || defined(__HAIKU__)) )
+ #define SUPPORT_XINPUT2 0
+ #elif ( (defined(__MACH__)) && (defined(__APPLE__)) )
+ #define SUPPORT_XINPUT2 0
+diff --git a/src/fs-uae/main.c b/src/fs-uae/main.c
+index f3f0370..6aab284 100644
+--- a/src/fs-uae/main.c
++++ b/src/fs-uae/main.c
+@@ -40,6 +40,10 @@ static int fs_uae_argc;
+ static char **fs_uae_argv;
+ static int g_warn_about_missing_config_file;
+ 
++#ifdef __HAIKU__
++int log_bsd = 0;
++#endif
++
+ #define LOG_LINE "---------------------------------------------------------" \
+         "-------------------\n"
+ 
+diff --git a/src/scp.cpp b/src/scp.cpp
+index 0d7a6a5..7d1793f 100644
+--- a/src/scp.cpp
++++ b/src/scp.cpp
+@@ -21,6 +21,17 @@
+ 
+ #define MAX_REVS 5
+ 
++#ifdef __HAIKU__
++#include <stdint.h>
++#include <posix/endian.h>
++#include <support/ByteOrder.h>
++
++#define __bswap16(x)	__swap_int16(x)
++#define	bswap16(x)	__bswap16(x)
++#define	be16toh(x)	bswap16((x))
++#define	le32toh(x)	((uint32_t)(x))
++#endif
++
+ enum pll_mode {
+     PLL_fixed_clock, /* Fixed clock, snap phase to flux transitions. */
+     PLL_variable_clock, /* Variable clock, snap phase to flux transitions. */
+-- 
+2.13.1
+
+
+From 4b1e880984732365c3b13e01b6908bc738d7dbd2 Mon Sep 17 00:00:00 2001
+From: Chris Moore <chris@mooreonline.org>
+Date: Mon, 7 Aug 2017 19:13:30 +0100
+Subject: Further fixes
+
+
+diff --git a/src/ethernet.cpp b/src/ethernet.cpp
+index 115aa5a..d2f2963 100644
+--- a/src/ethernet.cpp
++++ b/src/ethernet.cpp
+@@ -14,7 +14,7 @@
+ #include "uae/slirp.h"
+ 
+ #ifndef HAVE_INET_ATON
+-static int inet_aton(const char *cp, struct in_addr *ia)
++int inet_aton(const char *cp, struct in_addr *ia)
+ {
+ 	uint32_t addr = inet_addr(cp);
+ 	if (addr == 0xffffffff)
+diff --git a/src/jit/exception_handler.cpp b/src/jit/exception_handler.cpp
+index 8cf96f8..2e8b873 100644
+--- a/src/jit/exception_handler.cpp
++++ b/src/jit/exception_handler.cpp
+@@ -9,8 +9,10 @@
+ #ifndef __USE_GNU
+ #define __USE_GNU
+ #endif
++#ifndef __HAIKU__
+ #include <sys/ucontext.h>
+ #endif
++#endif
+ #include <signal.h>
+ 
+ #define SIG_READ 1
+-- 
+2.13.1
+
+
+From 20c1f1fc01eaa38ca0dd57fdca8060f2839150b2 Mon Sep 17 00:00:00 2001
+From: Chris Moore <chris@mooreonline.org>
+Date: Mon, 7 Aug 2017 19:59:13 +0100
+Subject: Fix dat location
+
+
+diff --git a/libfsemu/src/data.c b/libfsemu/src/data.c
+index 2d597ad..26f8198 100644
+--- a/libfsemu/src/data.c
++++ b/libfsemu/src/data.c
+@@ -288,6 +288,8 @@ int fs_data_init(const char *app_name, const char *dat_name)
+     if (result != 1) {
+         return 1;
+     }
++    
++    char* data_path = "/system/data/fs-uae";
```

Comment:
This shouldn't be hardcoded. here is a good example: https://github.com/haikuports/haikuports/blob/master/games-action/teeworlds/patches/teeworlds_0.6.3.patchset#L337